### PR TITLE
tx pool callback

### PR DESCRIPTION
**Pull request by zhangsoledad**

### CODE REVIEW

File: util/types/src/core/service.rs [hunk 1]
```diff
@@ -29,3 +31,57 @@ impl<A, R> Request<A, R> {
...
+    pub cycles: Cycle,
+    /// Transaction serialized cycles
+    pub size: usize,
+    /// Transaction fee
+    pub fee: Capacity,
+}
+
+/// TX reject message
+#[derive(Debug, Clone, PartialEq, Eq, Hash, Serialize, Deserialize)]
+pub enum PoolTransactionReject {
```
Comment (yangby-cryptape) [inline util/types/src/core/service.rs:50]:
```suggestion
#[derive(Debug, Clone, PartialEq, Eq, Hash, Serialize, Deserialize)]
#[serde(tag = "type", content = "description")]
pub enum PoolTransactionReject {
```

So, the return looks like the follow structure:
```javascript
[
    // first item is the `PoolTransactionEntry`
    {
        "transaction": { ... /* omit */ ... }
    },
    // second item is the `PoolTransactionReject`
    {
        "type": "Resolve",
        "description": "Resolve failed Dead(OutPoint(...))"
    }
]  
```

**Comment by yangby-cryptape** [comment]
If a proposed transaction doesn't be committed during the proposed window, subscribers couldn't know that.

**Comment by yangby-cryptape** [comment]
I have tested, both methods work well.

**Comment by zhangsoledad** [comment]
benchmark

**Comment by nervos-bot-user** [comment]
#### Benchmark Result
  - TPS: 491
  - CKB Version: dfc7a4d11e86f86f638b2582248d352ce014316c
  - Instance Type: c5.xlarge
  - Instances Count: 3
  - Bench Type: In2Out2
  - CKB Logger Filter: info,ckb=debug

**Review by quake — APPROVED** [review]

File: rpc/src/module/subscription.rs [hunk 2]
```diff
@@ -315,6 +314,40 @@ impl SubscriptionRpcImpl {
                             break;
                         },
                     },
+                    recv(proposed_transaction_receiver) -> msg => match msg {
```
Comment (doitian) [inline rpc/src/module/subscription.rs:317]: Please document new subscription topics in `SubscriptionRpc::subscribe`.

**Review by doitian — CHANGES_REQUESTED** [review]

**Review by doitian — DISMISSED** [review]
I'll fix the RPC doc link later

**Comment by doitian** [comment]
Test failed:

```
thread 'tests::block_assembler::test_block_template_timestamp' panicked at 'assertion failed: `(left == right)`
  left: `1608657504170`,
 right: `11`', chain/src/tests/block_assembler.rs:157:5
```

File: notify/src/lib.rs [hunk 3]
```diff
@@ -262,6 +344,34 @@ impl NotifyController {
         let _ = self.new_transaction_notifier.send(tx_entry);
     }
 
+    /// TODO(doc): @quake
+    pub fn subscribe_proposed_transaction<S: ToString>(
+        &self,
+        name: S,
+    ) -> Receiver<PoolTransactionEntry> {
+        Request::call(&self.proposed_transaction_register, name.to_string())
+            .expect("Subscribe new transaction should be OK")
```
Comment (keroro520) [inline notify/src/lib.rs:353]: panic messages should be "Subscribe proposed transaction ...".
And also `subscribe_reject_transaction`

**Review by keroro520 — CHANGES_REQUESTED** [review]

**Comment by keroro520** [comment]
LGTM
I have reviewed and tested this PR

**Review by keroro520 — APPROVED** [review]

**Review by keroro520 — APPROVED** [review]

**Review by doitian — APPROVED** [review]

**Comment by doitian** [comment]
bors r=doitian,keroro520

**Pull request merged**